File: Week2/fibonacci_last_digit/fibonacci_last_digit.cpp
#include <iostream>
#include <vector>
using namespace std;

int get_fibonacci_last_digit(int n) {
  vector<int> fib_list {0,1};
    if (n<=1){
     return fib_list[n];
    }
    for (int i = 2; i <= n; i++) {
     fib_list.push_back((fib_list[i-1] + fib_list[i-2]) % 10);
    }
  return fib_list[n];
}

int main() {
  int n;
  cin >> n;
  int c = get_fibonacci_last_digit(n);
  cout << c << '\n';
}

File: Week2/fibonacci_last_digit/fibonacci_last_digit.rb
def get_fibonacci_last_digit(n)
  fib_list = [0, 1]
  return fib_list[n] if n <= 1
  (2..n).each { |i| fib_list << (fib_list[i-1] + fib_list[i-2]) % 10 }
  fib_list[n]
end

n = gets.chomp.to_i
puts get_fibonacci_last_digit n